### PR TITLE
Fix stepper motor timeout

### LINKED ISSUE #237: Connection to ST10 times out
The problem seems to be that the `wait_until_stopped()` method is timing out waiting for a response from the device (to the "send string" command). Curiously, it seems that the response is actually being received by the program, but seemingly it is not being returned appropriately to `wait_until_stopped()`. Presumably some of the logic around the `_SerialReader` class is broken.

### PR DESCRIPTION
**Pull request by alexdewar**
Fixes #237.

It took a while for me to figure out what was going on here, but it turns out there were two problems.

Firstly, the homing routine wasn't waiting for long enough before timing out. We forgot that the timeout is for *all* of the homing movements, not just the last one, so I've extended it to 10s.

Secondly, the logic around timing out for synchronous reads was broken. As we do all the reading for the device on a separate thread in order to intercept asynchronous messages (viz. the "I've stopped moving" message) that can occur at some unspecified time, we need the read timeout for the serial device to be infinity. On the other hand, we don't want the synchronous reads to wait forever (in case the device has disconnected), but the solution isn't to do what I did before and temporarily change the read timeout; what happens in that case is that the reading thread will start a new read with this new timeout then raise an exception when no new message appears, even though no message is expected. Now the timeout is set to infinity and instead we check whether synchronous reads have timed out by adding a timeout for reading the messages from `_SerialReader`'s `out_queue`.

I also changed things to allow the user to set the log level with an environment variable.

### CODE REVIEW

**Comment by alexdewar** [comment]
@dalonsoa @dc2917 If either/both of you have a chance to review this today, that'd be great. I'm going into the lab tomorrow so ideally I'd like to test this branch after making any suggested changes if possible.

No biggie if you're busy though!

File: finesse/logger.py [hunk 1]
```diff
@@ -13,9 +14,15 @@ def initialise_logging() -> None:
     log_path.mkdir(parents=True, exist_ok=True)
     filename = log_path / f"{datetime.now().strftime('%Y%m%d_%H-%M-%S')}.log"
 
+    # Allow user to set log level with environment variable
+    log_level = os.environ.get("FINESSE_LOG_LEVEL") or "INFO"
+    log_level = log_level.upper()
```
Comment (dalonsoa) [inline finesse/logger.py:19]:
```suggestion
    log_level = (os.environ.get("FINESSE_LOG_LEVEL") or "INFO").upper()
```
I've never been a big fan of assigning a variable just to change immediately afterwards.

**Review by dalonsoa — APPROVED** [review]
Looks good! Let's hope this works.

**Review by dc2917 — APPROVED** [review]
Looks good to me!

**Comment by alexdewar** [comment]
Thanks both. I'll hold off on merging this until I've tested it with the real hardware.

**Pull request merged**